### DISCUSSION TITLE
Creating an Invoice

### DISCUSSION THREAD

**Discussion by HaHa-Clint** [Budibase/budibase#11728] [Help]
Good Afternoon, 

I'm recently new to the Budibase platform. Is there anyone that can help with a tutorial or example creating an invoice for a new customer?

Thank you for the help!

**Comment by ConorWebb96** [comment]
Hey @HaHa-Clint,

I believe there is a template within Budibase which you could use thats based around invoices? I've supplied a screenshot of this below:
<img width="1260" alt="Screenshot 2023-09-11 at 14 14 36" src="https://github.com/Budibase/budibase/assets/126772285/51c710ce-c6f6-455e-9980-1acd95fef1e8">

To get to this page follow these steps;
1. Click the `Create new app` button.
2. Find the `invoice approval software` template.
3. Hover your cursor over it and click `Use template`
4. Change the name or just click `Create app`.

I hope this helps

**Reply by HaHa-Clint** [reply]
@ConorWebb96 Thank you for your help. This is useful, but It does not solve my problem entirely.  

With my project, I am working with an insurance company. We need a way to create or generate an invoice for when a customer has applied for certain coverages. 

Is there a way that we could automate or auto-generate an invoice once a customer has determine what coverages they need?

Thank you for the help. 

I should have been a little more clear with what I was looking for in initially.

**Reply by ConorWebb96** [reply]
Hey @HaHa-Clint,

Theres a few ways you likely could go about doing this. 

You could create an automation using the `app action trigger` step, this would send out an email to your customers whenever they complete their final step of your insurance form. The action attached to your final submit button would be the `trigger automation action`.

You would likely would want to send whatever uniquely identifies these records along with the `trigger automation action`. As from here you can query the relevant rows and get all relevant data you wish to send via email. I've linked the docs that may help you in this case: `app action trigger` https://docs.budibase.com/docs/app-action and the trigger `automation action`  https://docs.budibase.com/docs/application-actions#trigger-automations

The other option you have is directing them to a separate page, which will have all the information pulled in based on the choices they have made. From here I guess you could take it a step further and possibly write a plugin that will convert the pages HTML to a PDF which they could then download. If you aren't confident with this, then displaying the information on the page is your next best option.

I hope this helps.
